src/common/get-card-selection-view-data.ts

```diff
@@ -1,65 +1,147 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-import { flatMap, forOwn, keys } from 'lodash';
+import { HighlightState } from 'common/components/cards/instance-details-footer';
+import { GetUnavailableHighlightStatus } from 'common/get-unavailable-highlight-status';
+import { UnifiedScanResultStoreData } from 'common/types/store-data/unified-data-interface';
+import { flatMap, forOwn, isEmpty, keys } from 'lodash';
+
 import {
     CardSelectionStoreData,
     RuleExpandCollapseData,
     RuleExpandCollapseDataDictionary,
 } from './types/store-data/card-selection-store-data';
 
 export interface CardSelectionViewData {
-    highlightedResultUids: string[]; // page elements to highlight
     selectedResultUids: string[]; // indicates selected cards
     expandedRuleIds: string[];
     visualHelperEnabled: boolean;
+    resultsHighlightStatus: ResultsHighlightStatus;
 }
 
-export type GetCardSelectionViewData = (storeData: CardSelectionStoreData) => CardSelectionViewData;
+export type ResultsHighlightStatus = { [resultUid: string]: HighlightState };
 
-export const getCardSelectionViewData: GetCardSelectionViewData = (
+export type GetCardSelectionViewData = (
     storeData: CardSelectionStoreData,
+    unifiedScanResultStoreData: UnifiedScanResultStoreData,
+    getUnavailableHighlightStatus: GetUnavailableHighlightStatus,
+) => CardSelectionViewData;
+
+export const getCardSelectionViewData: GetCardSelectionViewData = (
+    cardSelectionStoreData: CardSelectionStoreData,
+    unifiedScanResultStoreData: UnifiedScanResultStoreData,
+    getUnavailableHighlightStatus: GetUnavailableHighlightStatus,
 ): CardSelectionViewData => {
     const viewData = getEmptyViewData();
 
-    if (!storeData) {
+    if (
+        isEmpty(cardSelectionStoreData) ||
+        isEmpty(unifiedScanResultStoreData) ||
```

Comment:
I think can be simplified a little (and reduced to one line) by replacing `isEmpty(unifiedScanResultStoreData) || unifiedScanResultStoreData.results == null` with `unifiedScanResultStoreData?.results == null`